### ISSUE TITLE
2021-12-27

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#696]
56P46, Flow-based generative model; 最大似然估计==最小KL散度; flow-based model可以直接maximize likelihood; Jacobian matrix; determinant; 行列式是高维空间中的体积; change of variable theorem; p(x’)*|det(Jf)| = pai(z’); dim(z) = dim(x); 多个G叠加; 训练的实际是G-1; coupling layer; 交替使用; 1*1 convolution; W can shuffle channels; (2021-12-18)
57P47, HW6; (2021-12-18)
58P48, HW6, English version; (2021-12-19)
59
60Chapter 7, 自监督学习; 
61P49, 芝麻街与进击的巨人; 自监督模型都是以芝麻街命名; BERT是非常巨大的模型, 340 million参数; (2021-12-19)
62P50, BERT简介; self-supervised learning; 一部分作为模型的输入, 另一部分作为模型的标注; 没有label就是unsupervised learning; masking input; transformer encoder就是Bert架构, Bert训练的是encoder; randomly mask/replace some tokens; 输出接近被遮挡的word; next sentence prediction; 判断两个句子是否相接; 这一招不太有用, 可能因为此任务太简单了; sentence order prediction; downstream task可能和天空没有关系; pre-train->fine-tune; GLUE scores; how to use Bert? Case 1, 判断句子正面还是负面; pre-train better than random; Case 2, 词性标注; Case 3, 自然语言推断; case 4, 问答(答案一定在文章里), 找出正确答案在文章中的起始位置和结束位置; training Bert is expensive; BERT胚胎学;  pre-training a seq2seq model; MASS/BART; T5 - comparison; (2021-12-19)
63P51, Bert的奇闻异事; why does Bert work? Embedding vector代表了word的意思; similar embedding has similar meaning, context is being considered; You shall know a word by the company it keeps; Bert是复杂版本的CBOW, 而且Bert还可以考虑上下文, 所以Bert抽取的word是contextualized word embedding; apply Bert to protein&DNA, 效果也很好; cross-lingual alignment; 只有资料量足够大才有效; 语言差异可以用向量表示; (2021-12-19)
64P52, GPT的野望; predict next token; they can do generation; how to use GPT? Few-shot learning, no gradient descent; one-shot learning; zero-shot learning; beyond text; Image-simCLR; BYOL; speech; speech GLUE; (2021-12-19)
65P53, auto-encoder上, 基本概念; auto-encoder类似self-supervised, 不需要人为标注书签; 使encoder输入和decoder输出越接近越好, 类似于cycle GAN; embedding == representation == code (low dim, bottleneck); dimension reduction; why auto-encoder? 化繁为简, 抽象出最本质的东西; de-noising auto-encoder; 还原加入noise之前的结果; bert的mask可以视为noise; Bert == de-noising auto-encoder; (2021-12-19)
66P54, auto-encoder下, 领结变声器与更多应用; feature disentangle; 对向量维数代表的内容进行区分; application, voice conversion; 把我的声音转成新垣结衣的声音; discrete latent representation; 强迫embedding vector为one-hot; VQVAE; text as representation; embedding的不是向量, 而是文字; 或许文字就是文章的摘要; 其实不是, 再加上GAN使之合乎文法; 也是cycle GAN的另一种形式; tree as embedding; decoder可以视为generator; 用来compress; anomaly detection; input is similar or not; 用来判别欺诈; (2021-12-19)
67P55, Bert and its family; pre-train & fine-tune; 通过大量语料训练语感, 再做少量题就可以了; pre-train model; represent each token by an embedding vector; 这种token不考虑上下文; English word as token; contextualized word embedding; LSTM, self-attention layers, tree-based model(擅长处理文法结构严谨的, 比如数学表达式); bigger model; smaller model; Albert; network compression; how to fine-tune; w1w2[sep]w3w4w5; 看到CLS要产生和整个句子有关的token; output: (1) class for each token; (2) copy from input(extraction-based QA); (3) general sequence; only fine-tune task specific or fine-tune all (better); only fine-tune adaptor part; weighted features; (2021-12-19)
